src/templating/templating.js

```diff
@@ -224,7 +224,7 @@
             if (typeof bindingValue == "string" || bindingValue['name']) {
                 // It's a named template - clear the element
                 ko.virtualElements.emptyNode(element);
-            } else if ('nodes' in bindingValue) {
+            } else if (bindingValue['nodes']) {
```

Comment:
I will clean this up as soon as I have some time, but I guess I don't need to rush since there are tons of PRs sitting here.
Anyway I had a situation similar to this:

```
var model = {
  dataTemplate: ko.observable(), // can be an ID string, HTML string or Node array
  data: ko.observable()
}

...

function getTemplateNodes(template) {
  if (template && template.length) return template;
  if ("template is HTML string")
    return ko.utils.parseHtml(template);
  return null;
}

...

<div data-bind="template: { name: isTemplateName(dataTemplate()) ? dataTemplate() : null, nodes: getTemplateNodes(dataTemplate()) }"></div>
```

With the current implementation KO will always use the `nodes` binding whenever it's presented in the bound object, regardless of what it returns. I needed here a way to be able to have a name or have `nodes` conditionally based on the `dataTemplate` observable.

I checked several history entries to have an idea wether the current behavior is be intention or not, but haven't found any.

I will put together a separate PR for it for discussion. For now I extended the `template` binding to reach the proposed behavior.